cmd/lvmd/app/root.go

```diff
@@ -122,6 +124,21 @@ func subMain(parentCtx context.Context) error {
 		}()
 	}
 
+	var metricsServer *http.Server
+	if metricsBindAddress != "" {
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", promhttp.Handler())
+		metricsServer = &http.Server{
+			Addr:    metricsBindAddress,
+			Handler: mux,
+		}
+		go func() {
+			if err := metricsServer.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
```

Comment:
What happens if the metrics Server doesnt shut down in time of a SIGTERM signal? I think we could bind this go routine to a context if possible. (Similar to the pprof endpoint below with a Shutdown loagic on the parent Ctx). Since this is quite some boilerplate it might make sense to start extracting these starts out into separate functions and using something like a wait group to orchestrate them all together.

In the current form, the start method is getting a little too long and its becoming hard to read due to different start / stop styles

Reply:
@jakobmoellerdev Ok, I've (at least) made the main startup method for lvmd shorter from my recent changes, but I hesitated to start extracting out pieces for starting these common http servers as I feel I'd be copying around the `sync.WaitGroup`, which seems to be an anti-pattern. The only other way I see to do this is to return a func from some other start* func, which is then called in a goroutine, using the waitgroup.

example
```
wd.Add(1)
fn := startHTTPServer(server)
go func() {
  defer wg.Done()
  fn()
}()
```

Any thoughts on the approach I've taken, vs what you were thinking? Thanks for any input before I go further down this path.